### PR TITLE
Parsers

### LINKED ISSUE #1: Parse buildinfo file
Create a script under `scripts/ingestion` to handle the debian project's `.buildinfo` file format  (spec [here](https://wiki.debian.org/ReproducibleBuilds/BuildinfoFiles)).

The output format should at a minimum follow the following schema:

` source | package | architecture | version | dependencies (as a list of packages) `

- [x] Parse .buildinfo file
- [x] Test correctness against package repositories
- [x] Handle split package case
   - Identify split packages
   - Create one entry for each split package

### PR DESCRIPTION
**Pull request by SahithiKasim**
closes #1

### CODE REVIEW

File: scripts/ingestion/database/buildinfo_db_init.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 import sqlite3
 
-conn = sqlite3.connect('/data/yellow/vineet/database/bi_multi_tables.db')
-cur = conn.cursor()
+def target_con(con):
+       conn = sqlite3.connect(con)
+       cur = conn.cursor()
+       return cur
 
+target_con('/data/yellow/vineet/database/bi_multi_tables.db')
```
Comment (SantiagoTorres) [inline scripts/ingestion/database/buildinfo_db_init.py:9]: can we update this parameter to be a constant/variable that can be adjusted? something like `DATABASE_LOCATION` would be good

File: scripts/ingestion/database/maintainer_db_init.py [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
 import sqlite3
 
-conn = sqlite3.connect('/data/yellow/vineet/database/bi_multi_tables.db')
-cur = conn.cursor()
+def target_con(con):
+        conn = sqlite3.connect(con)
+        cur = conn.cursor()
+        return cur
+
+target_con('/data/yellow/vineet/database/bi_multi_tables.db')
```
Comment (SantiagoTorres) [inline scripts/ingestion/database/maintainer_db_init.py:9]: same as above, you amy want to use the same variable (perhaps under a separate file called `constants.py`)

File: scripts/ingestion/database/popcon_db_init.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import sqlite3
+
+def target_con(con):
+        connection = sqlite3.connect(con)
+        cursor = connection.cursor()
+        return cursor
+
+def db_init_main():
+        target_con('/data/yellow/vineet/database/bi_multi_tables.db')
```
Comment (SantiagoTorres) [inline scripts/ingestion/database/popcon_db_init.py:9]: note on the same constant

File: scripts/ingestion/parsers/popcon_db_init.py [hunk 4]
```diff
@@ -1,20 +0,0 @@
-import sqlite3
```
Comment (SantiagoTorres) [inline scripts/ingestion/parsers/popcon_db_init.py:1]: is this file not needed anymore?

File: scripts/ingestion/popcon_main.py [hunk 5]
```diff
@@ -11,7 +12,8 @@
         x=datetime.now()
         for row in data:
                 name=row[1]
-                date="09/06/2022"
+                #date="09/06/2022"
```
Comment (SantiagoTorres) [inline scripts/ingestion/popcon_main.py:15]: please remove this commented-out line

**Review by SantiagoTorres — CHANGES_REQUESTED** [review]

File: scripts/ingestion/database/buildinfo_db_init.py (hunk 1 above)
Comment (SahithiKasim) [reply scripts/ingestion/database/buildinfo_db_init.py]: Declared it as constant in constants.py

File: scripts/ingestion/database/maintainer_db_init.py (hunk 2 above)
Comment (SahithiKasim) [reply scripts/ingestion/database/maintainer_db_init.py]: declared it as constant under constants.py

File: scripts/ingestion/parsers/popcon_db_init.py (hunk 4 above)
Comment (SahithiKasim) [reply scripts/ingestion/parsers/popcon_db_init.py:1]: We may need it for future references

File: scripts/ingestion/database/popcon_db_init.py (hunk 3 above)
Comment (SahithiKasim) [reply scripts/ingestion/database/popcon_db_init.py]: Declared it as constant

File: scripts/ingestion/popcon_main.py (hunk 5 above)
Comment (SahithiKasim) [reply scripts/ingestion/popcon_main.py]: Removed it.

File: scripts/ingestion/database/maintainer_db_init.py (hunk 2 above)
Comment (SahithiKasim) [reply scripts/ingestion/database/maintainer_db_init.py]: Declared

**Comment by SahithiKasim** [comment]
I cannot merge the changes until you address the issue. Can you please look into it?

File: scripts/ingestion/constants.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+db_loc = '/data/yellow/vineet/database/bi_multi_tables.db'
+maint_inst_loc = "https://popcon.debian.org/maint/by_inst"
+maintainer_tar_file = '/data/yellow/vineet/python_files/new_scripts/database_creation/maintainer.txt'
+regex = "(\ )+"
```
Comment (SantiagoTorres) [inline scripts/ingestion/constants.py:7]: please use uppercase for connstants

File: scripts/ingestion/maintainer_main.py [hunk 7]
```diff
@@ -1,11 +1,9 @@
 #!/usr/bin/env python
 import csv
 import sqlite3
+from scripts.ingestion.database.maintainer_db_init import db_init_main
 
-conn = sqlite3.connect('/data/yellow/vineet/database/bi_multi_tables.db')
-cur = conn.cursor()
-
-def main():
+def maint():
```
Comment (SantiagoTorres) [inline scripts/ingestion/maintainer_main.py:6]: why is this called maint now?

File: scripts/ingestion/parsers/popcon_compare.py [hunk 8]
```diff
@@ -7,18 +7,17 @@
 import csv
 import os
 import shutil
+from scripts.ingestion.constants import *
```
Comment (SantiagoTorres) [inline scripts/ingestion/parsers/popcon_compare.py:10]: please don't use `*` imports. It's hard to know what variables are being [shadowed](https://fixpython.com/2022/08/fix-python-what-is-the-problem-with-shadowing-names-defined-in-outer-scopes-0ja9koj/)

File: scripts/ingestion/parsers/popcon_parser.py [hunk 9]
```diff
@@ -6,15 +6,14 @@
 import csv
 import os
 import shutil
+from scripts.ingestion.constants import *
```
Comment (SantiagoTorres) [inline scripts/ingestion/parsers/popcon_parser.py:9]: same here

File: scripts/ingestion/database/maintainer_db_init.py [hunk 10]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
+import sqlite3
+from scripts.ingestion.constants import *
```
Comment (SantiagoTorres) [inline scripts/ingestion/database/maintainer_db_init.py:3]: `*` import

File: scripts/ingestion/database/buildinfo_db_init.py [hunk 11]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python
+import sqlite3
+from scripts.ingestion.constants import *
```
Comment (SantiagoTorres) [inline scripts/ingestion/database/buildinfo_db_init.py:3]: `*` import

File: scripts/ingestion/parsers/popcon_compare.py [hunk 12]
```diff
@@ -58,7 +57,6 @@
         print("Downloaded Data")
     
 fdin.close()
-totinst=282358448
 
 # Copying the data from text file to csv 
 with open('/data/yellow/vineet/python_files/new_scripts/data_pre_processing/today.txt', 'r', encoding="utf-8") as fdin, open('/data/yellow/vineet/python_files/new_scripts/data_pre_processing/today.csv','w', encoding="utf-8",newline='') as fdout:
```
Comment (SantiagoTorres) [inline scripts/ingestion/parsers/popcon_compare.py:62]: this is still using some magic constants here

File: scripts/ingestion/popcon_main.py [hunk 13]
```diff
@@ -0,0 +1,29 @@
+import csv
+from datetime import datetime
+import sqlite3
+from scripts.ingestion.database.popcon_db_init import db_init_main
+
+def popcon():
+        with open('/data/yellow/vineet/python_files/new_scripts/data_pre_processing/today.csv', 'r',encoding= 'unicode_escape') as file:
```
Comment (SantiagoTorres) [inline scripts/ingestion/popcon_main.py:7]: magic value again

**Review by SantiagoTorres — CHANGES_REQUESTED** [review]
I think we could improve things a bit still

**Comment by SahithiKasim** [comment]
I made the changes. Can you check them?

File: scripts/ingestion/constants.py (hunk 6 above)
Comment (SahithiKasim) [reply scripts/ingestion/constants.py]: Changed to Uppercase

**Comment by SahithiKasim** [comment]
Can you recheck the scripts and tell me the changes?

File: scripts/ingestion/parsers/popcon_parser.py [hunk 14]
```diff
@@ -1,48 +1,58 @@
...
+        print(e)
+
+    # Replace separator in line with commas
+    comma_divided = re.sub(REGEX, ",", rest)
+    rank, name, inst, vote, old, recent, no_files, _ = \
+            comma_divided.split(",")
+
+    return (rank, name, inst, vote, old, recent, no_files, maintainer)
+
+def parser():
```
Comment (SantiagoTorres) [inline scripts/ingestion/parsers/popcon_parser.py:32]: set file as parameter please.

Comment (SantiagoTorres) [reply scripts/ingestion/parsers/popcon_parser.py:32]:
```
def parser(filename = OUTPUT_FILE)
```

File: scripts/ingestion/popcon_main.py [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+                        no_files=row[6]
+                        maintainer=row[7]
+                        cursor.execute('''INSERT OR REPLACE INTO popularity_table(name,date,inst,vote,old,recent,no_files,maintainer) VALUES (?,?,?,?,?,?,?,?)''',(name,date,inst,vote,old,recent,no_files,maintainer))
+                        conn.commit()
+                        no_records += 1
+
+if __name__ == "__main__":
+        db_init()
+        parser()
+        popcon()
```
Comment (SantiagoTorres) [inline scripts/ingestion/popcon_main.py]:
```
        handle = db_init()
        filename = parser(handle)
        popcon(filename)
```

**Review by SantiagoTorres — APPROVED** [review]

File: scripts/ingestion/popcon_main.py (hunk 15 above)
Comment (SahithiKasim) [reply scripts/ingestion/popcon_main.py]: Updated the scripts

**Pull request merged**